File: tests/validate_livestock_inventory.py
#!/usr/bin/env python3

import json
import sys
from pathlib import Path


path = Path(sys.argv[1] if len(sys.argv) > 1 else "inventory/livestock_reconciliation.json")
report = json.loads(path.read_text(encoding="utf-8"))


def expect(value: bool, message: str) -> None:
    if not value:
        raise AssertionError(message)


public = report["public_release"]
alignment = report["alignment"]
identity = report["identity_integrity"]
privacy = report["privacy"]

expect(public["doi"] == "10.7910/DVN/75E7HV", "Unexpected DOI")
expect(public["version"] == "2.0", "Unexpected release version")
expect(
    public["md5"] == "9dd9b11879805f22c18ec7e0173f80ba",
    "Public file checksum changed",
)
expect(public["rows"] == 2503, "Public row count changed")
expect(public["meaningful_columns"] == 38, "Column count changed")
expect(alignment["aom_id_mismatches"] == 0, "AOM IDs drifted")
expect(alignment["hierarchy_level_mismatches"] == 0, "Hierarchy drifted")
expect(identity["duplicate_aom_ids"] == "AOM_006275", "Duplicate-ID case changed")
expect(not privacy["workbook_path_published"], "Private workbook path exposed")
expect(
    not privacy["workbook_fingerprint_published"],
    "Private workbook fingerprint exposed",
)
expect(not privacy["ssa_feedsdb_values_published"], "Restricted SSA Feeds values exposed")

print("Livestock inventory validation passed")

File: tests/validate_livestock_inventory.R
#!/usr/bin/env Rscript

path <- if (length(commandArgs(trailingOnly = TRUE))) {
  commandArgs(trailingOnly = TRUE)[[1]]
} else {
  "inventory/livestock_reconciliation.json"
}

if (!requireNamespace("jsonlite", quietly = TRUE)) {
  stop("Missing R package: jsonlite", call. = FALSE)
}

report <- jsonlite::read_json(path, simplifyVector = TRUE)
expect <- function(value, message) {
  if (!isTRUE(value)) stop(message, call. = FALSE)
}

expect(report$public_release$doi == "10.7910/DVN/75E7HV", "Unexpected DOI")
expect(report$public_release$version == "2.0", "Unexpected release version")
expect(
  report$public_release$md5 == "9dd9b11879805f22c18ec7e0173f80ba",
  "Public file checksum changed"
)
expect(report$public_release$rows == 2503L, "Public row count changed")
expect(report$public_release$meaningful_columns == 38L, "Column count changed")
expect(report$alignment$aom_id_mismatches == 0L, "AOM IDs drifted")
expect(report$alignment$hierarchy_level_mismatches == 0L, "Hierarchy drifted")
expect(
  report$identity_integrity$duplicate_aom_ids == "AOM_006275",
  "Duplicate-ID case changed"
)
expect(
  report$privacy$workbook_path_published == FALSE,
  "Private workbook path exposed"
)
expect(
  report$privacy$workbook_fingerprint_published == FALSE,
  "Private workbook fingerprint exposed"
)
expect(
  report$privacy$ssa_feedsdb_values_published == FALSE,
  "Restricted SSA Feeds values exposed"
)

cat("Livestock inventory validation passed\n")
